config.go

```diff
@@ -0,0 +1,226 @@
+	if err != nil {
+		return Config{}, err
+	}
+	config.AuthCredentials = authConfig
+
+	// Split lane option
+	laneOptions, err := shellquote.Split(config.Lane)
+	if err != nil {
+		return Config{}, fmt.Errorf("Failed to parse lane (%s), error: %s", config.Lane, err)
+	}
+	config.LaneOptions = laneOptions
+
+	return config, nil
+}
+
+func (s FastlaneRunner) validateAuthInputs(config Config) (appleauth.Inputs, error) {
+	authInputs := appleauth.Inputs{
+		Username:            config.AppleID,
+		Password:            string(config.Password),
+		AppSpecificPassword: string(config.AppSpecificPassword),
+		APIIssuer:           config.APIIssuer,
+		APIKeyPath:          string(config.APIKeyPath),
+	}
+	if err := authInputs.Validate(); err != nil {
+		return appleauth.Inputs{}, fmt.Errorf("Issue with authentication related inputs: %v", err)
+	}
+	return authInputs, nil
+}
+
+func (s FastlaneRunner) parseAuthSources(bitriseConnection string) ([]appleauth.Source, error) {
+	switch bitriseConnection {
+	case "automatic":
+		return []appleauth.Source{
+			&appleauth.ConnectionAPIKeySource{},
+			&appleauth.ConnectionAppleIDFastlaneSource{},
+			&appleauth.InputAPIKeySource{},
+			&appleauth.InputAppleIDFastlaneSource{},
+		}, nil
+	case "api_key":
+		return []appleauth.Source{&appleauth.ConnectionAPIKeySource{}}, nil
+	case "apple_id":
+		return []appleauth.Source{&appleauth.ConnectionAppleIDFastlaneSource{}}, nil
+	case "off":
+		return []appleauth.Source{
+			&appleauth.InputAPIKeySource{},
+			&appleauth.InputAppleIDFastlaneSource{},
+		}, nil
+	default:
+		return nil, fmt.Errorf("invalid connection input: %s", bitriseConnection)
+	}
+}
+
+func (s FastlaneRunner) validateGemHome(config Config) {
+	if strings.TrimSpace(config.GemHome) == "" {
+		return
+	}
+	s.logger.Warnf("Custom value (%s) is set for GEM_HOME environment variable. This can lead to errors as gem lookup path may not contain GEM_HOME.")
+}
+
+func (s FastlaneRunner) getWorkdir(config Config) (string, error) {
+	s.logger.Infof("Expand WorkDir")
+
+	workDir := config.WorkDir
+	if workDir == "" {
+		s.logger.Printf("WorkDir not set, using CurrentWorkingDirectory...")
+		currentDir, err := pathutil.CurrentWorkingDirectoryAbsolutePath()
+		if err != nil {
+			return "", fmt.Errorf("Failed to get current dir, error: %s", err)
+		}
+		workDir = currentDir
+	} else {
+		absWorkDir, err := pathutil.AbsPath(workDir)
+		if err != nil {
+			return "", fmt.Errorf("Failed to expand path (%s), error: %s", workDir, err)
+		}
+		workDir = absWorkDir
+	}
+
+	s.logger.Donef("Expanded WorkDir: %s", workDir)
+	return workDir, nil
+}
+
+func (s FastlaneRunner) checkForRbenv(workDir string) {
+	if _, err := s.cmdLocator.LookPath("rbenv"); err != nil {
+		cmd := s.rbyFactory.Create("rbenv", []string{"versions"}, &command.Opts{
+			Stderr: os.Stderr,
+			Stdout: os.Stdout,
+			Dir:    workDir,
+		})
+
+		fmt.Println()
+		s.logger.Donef("$ %s", cmd.PrintableCommandArgs())
+		if err := cmd.Run(); err != nil {
```

Comment:
When `err` is not nil, it usually just contains `exit status X`, while the real error is printed to the standard output of the command. This is why we usually use `RunAndPrintCombinedOutput()` instead of `Run()` and print the combined output when there is an error.

Reply:
After looking into this further it looks like if we use `RunAndReturnTrimmedCombinedOutput` (from `command` v2) an error will be thrown as it expects the stdout to not be set, but in this case we are setting it above this second of code.